creator-node/src/services/stateMachineManager/stateMonitoring/findReplicaSetUpdates.jobProcessor.js

```diff
@@ -303,23 +303,25 @@ const _findReplicaSetUpdatesForUser = async (
 }
 
 /**
- * Transforms data type from ((K1,V1) => (K2,V2)) to (K1,V1[K2]), where K1 is the node endpoint,
- * V1 is the mapping, and K2 is the wallet to filter by. Also filters out nodes that don't have a value for the wallet.
- * @param {Object} replicaSetNodesToUserClockStatusesMap map of secondary endpoint strings to (map of user wallet strings to clock value of secondary for user)
- * @param {string} wallet the wallet to filter clock values for (other wallets will be excluded from the output)
- * @returns mapping of node endpoint (string) to clock value (number) on that node for the given wallet
+ * Filters input map to only user info for provided wallet, also filtering out nodes that have no clock value for provided wallet
+ * @param {Object} replicaToUserInfoMap map(secondary endpoint => map(user wallet => { clock, filesHash }))
+ * @param {string} wallet the wallet to filter for (other wallets will be excluded from the output)
+ * @returns map(replica (string) => { clock (number), filesHash (string) } ) mapping of node endpoint to user info on that node for the given wallet
  */
-const _transformAndFilterNodeToClockValuesMapping = (
-  replicaSetNodesToUserClockStatusesMap,
+const _transformAndFilterReplicaToUserInfoMap = (
+  replicaToUserInfoMap,
   wallet
 ) => {
   return Object.fromEntries(
-    Object.entries(replicaSetNodesToUserClockStatusesMap)
-      .map(([node, clockValueMapping]) => [
+    Object.entries(replicaToUserInfoMap) // [[replica, map(wallet => { clock, filesHash })]]
+      .map(([node, userInfoMap]) => [
         node,
-        clockValueMapping[wallet] || -1
+        {
+          ...userInfoMap[wallet],
+          clock: userInfoMap[wallet][clock] || -1 // default clock to -1 where not present
```

Comment:
wouldn't this look for the value of a variable with the name `clock` rather than the field `clock` on the possibly null object returned by `userInfoMap[wallet]`? I think it should be changed to `userInfoMap[wallet]?.clock`

Reply:
yep good catch, will fix